### ISSUE TITLE
🐛 expected `,` but instead found `default`

### ISSUE THREAD

**Issue by bjornharrtell** [biomejs/biome#4765]
### Environment information

```block
CLI:
  Version:                      1.9.4
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.12.2"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "yarn/1.22.22"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 false

Workspace:
  Open Documents:               0
```

### What happened?

Exporting aliased default as type results in parse error.

Example export syntax that causes the parse error:

export { type default as CrsMeta } from './crs-meta.js'

### Expected result

It should not cause parse error.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct
